frame/support/procedural/src/lib.rs

```diff
@@ -778,6 +781,243 @@ pub fn storage_alias(_: TokenStream, input: TokenStream) -> TokenStream {
+	derive_impl::derive_impl(
+		__source_path.into(),
+		attrs.into(),
+		input.into(),
+		custom_attrs.disambiguation_path,
+	)
+	.unwrap_or_else(|r| r.into_compile_error())
+	.into()
+}
+
+/// The optional attribute `#[pallet::default_config]` can be attached to a `Config` trait
+/// within your pallet to mark it as auto-derivable under test conditions via
+/// [`[#[derive_impl(..)]`](`macro@derive_impl`).
+///
+/// You may use [`#[pallet::no_default]`](`macro@no_default`) and
+/// [`#[pallet::constant]`](`macro@constant`) on individual trait items within your `Config`
+/// trait.
+///
+/// The [`#[pallet::no_default]`](`macro@no_default`) attribute allows you to specify that a
+/// particular trait item _cannot_ be used as a default when a test `Config` is derived using
+/// the [`#[derive_impl(..)]`](`macro@derive_impl`) attribute macro.
+#[proc_macro_attribute]
+pub fn default_config(_: TokenStream, _: TokenStream) -> TokenStream {
+	pallet_macro_stub()
+}
+
+/// The optional attribute `#[pallet::no_default]` can be attached to trait items within a
+/// `Config` trait impl that has [`#[pallet::default_config]`](`macro@default_config`) attached
+/// to it.
+///
+/// Attaching this attribute to a trait item ensures that that trait item will not be used as a
+/// default with the [`#[derive_impl(..)]`](`macro@derive_impl`) attribute macro.
+#[proc_macro_attribute]
+pub fn no_default(_: TokenStream, _: TokenStream) -> TokenStream {
+	pallet_macro_stub()
+}
+
+/// Attach this attribute to an impl statement that needs to be registered for use with the
+/// [`macro@derive_impl`] attribute macro.
+///
+/// You must also specify an ident as the only argument to the attribute. This ident will be
+/// used as the export name for this default config impl, and is the name you must provide to
+/// [`macro@derive_impl`] when you import this impl.
+///
+/// The ident you provide is exported at the root of the crate where it is defined, so the
+/// ident must be unique at the crate level so as not to collide with other items. This is
+/// because internally this attribute aliases to macro_magic's `#[export_tokens]` macro, which
+/// relies on crate-level `macro_rules!` exports to export tokens for use elsewhere.
+///
+/// ## Example
+///
+/// ```ignore
+/// pub struct ExampleTestDefaultConfig {}
+///
+/// #[register_default_impl(ExampleTestDefaultConfig)]
+/// impl DefaultConfig for ExampleTestDefaultConfig {
+/// 	type Version = ();
+/// 	type BlockWeights = ();
+/// 	type BlockLength = ();
+/// 	...
+/// 	type SS58Prefix = ();
+/// 	type MaxConsumers = frame_support::traits::ConstU32<16>;
+/// }
+/// ```
+///
+/// This macro acts as a thin wrapper around macro_magic's `#[export_tokens]`. See the docs
+/// [here](https://docs.rs/macro_magic/latest/macro_magic/attr.export_tokens.html) for more info.
+#[proc_macro_attribute]
+pub fn register_default_impl(attrs: TokenStream, tokens: TokenStream) -> TokenStream {
```

Comment:
My reasoning against this is this can be used outside of pallets, and the whole `pallet::whatever` thing breaks down when we aren't in a pallet outer macro (i.e. we'll have to make a module `frame_support::pallet` which will undoubtedly cause issues with existing instances of `use frame_suppor::*`, so I thought a regular export from `frame_support` would be appropriate. Technically there is nothing about this that is specific to pallets so